src/sentry/event_manager.py

```diff
@@ -875,6 +873,22 @@ def save(self, project_id, raw=False, assume_normalized=False):
 
         return event
 
+    def _get_event(self, project_id, event_id):
+        nodestore_sample_rate = options.get("store.nodestore-sample-rate")
+        use_nodestore = random.random() < nodestore_sample_rate
+
+        if use_nodestore:
```

Comment:
When you discuss with James the rollout process, please ensure you have reliable metrics about:
- how often we return an event here
- how long it takes to load an event when it exists 
- how long it takes to load an event when it does not exist. (The load we are putting on nodestore here is vastly skewed towards reads of elements that do not exist. I am not sure we have a good sense around the performance of this specific load pattern (several systems are a lot less efficient at trying to find an element that does not exist than loading an element whern it exists)).